lib/OpenQA/Setup.pm

```diff
@@ -13,20 +13,110 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
-# This package contains shared functions between WebAPI and WebSockets
-package OpenQA::ServerStartup;
-
+package OpenQA::Setup;
+use Mojo::Log;
+use Mojo::Home;
 use strict;
 use warnings;
+use Mojo::Base -base;
+use Sys::Hostname;
+use File::Spec::Functions 'catfile';
+use Mojo::File 'path';
 use Config::IniFiles;
 use db_profiler;
 use db_helpers;
 use OpenQA::Utils;
-use Mojo::File 'path';
 
-sub read_config {
-    my $app = shift;
+has config => sub { {} };
+
+has log => sub { Mojo::Log->new(handle => \*STDOUT, level => "debug"); };
+
+has home => sub { Mojo::Home->new($ENV{MOJO_HOME} || '/') };
+
+has mode => 'production';
+
+has log_name => 'scheduler';
```

Comment:
Please make it empty, it can cause confusion:
`has log_name;` would be sufficient